src/terminal/clear.lua

```diff
@@ -0,0 +1,114 @@
+--- Clear Module
+-- Provides functions to clear various parts of the terminal screen.
+-- @module clear
+
+local output = require "terminal.output"
+
+local M = {}
+
+-- Lazy loading terminal to avoid circular dependency
+local terminal
+local function get_terminal()
+  terminal = terminal or require("terminal")
+  return terminal
+end
```

Comment:
This was to be expected to happen. Since it is most likely going to happen in multiple places as we further split up the main module. I think we better solve this in `src/module/init.lua`.

The way to do it would be by storing that module table into `package.loaded` before loading the submodules.

So this code can just be changed to:
```suggestion
local terminal = require("terminal")
```

See the suggestion in `init.lua` on how to change it there.

Reply:
this is outdated now